transit_simulation/route.py

```diff
@@ -24,10 +29,49 @@ def fetch_route(bus_route_id, direction, d_date, d_time):
     
     return route
 
+def fetch_HSL_gtfs_shapes():
+    '''Returns:
+    shape_df: A Pandas dataframe containing the information from the shapes.txt
+        file of a GTFS package.
+    '''
+    url = 'http://dev.hsl.fi/gtfs/hsl.zip'
+    
+    r = requests.get(url)
+    gtfs_zip = ZipFile(BytesIO(r.content))
+    shape_df = pd.read_csv(gtfs_zip.open('shapes.txt'))
+    
+    return shape_df
+
+def process_gtfs_shapes(shapes_data):
+    '''Args:
+    shapes_data: A pandas DataFrame containing data of shapes.txt in a gtfs package.
+        Practically route ids and points along the route.
+    
+    Returns:
+    routes: A GeoPandas GeoDataFrame containing unique route ids and the corresponding
+        route LineString as geometry.
+    '''
+    geodata = gpd.GeoDataFrame(shapes_data, geometry=gpd.points_from_xy(shapes_data.shape_pt_lat, shapes_data.shape_pt_lon))
+    routes = geodata.groupby(['shape_id'])['geometry'].apply(lambda x: LineString(x.tolist()))
+    return routes
+
+
 if __name__ == "__main__":
```

Comment:
This routine in the main looks like it's essentially a unit test.

it could be moved to `tests/test_route.py` as something like 

```python

import transit_simulation.route as route

def test_fetch_route():
        route_id = 1071
        direction = 1
        d_date = "2020-04-29"
        d_time = 16800

        pointlist = route.fetch_route(route_id, direction, d_date, d_time)
        print(pointlist)

def test_gtfs_shapes():
    ...
```

So that `pytest` can find them and run them as part of the test suite. It finds test functions based on file name and function name, so the `test_` in the beginning of the names is the important part.

Reply:
Thanks for this, I'll do these changes!